src/api.js

```diff
@@ -134,6 +134,30 @@ export const authenticatedRequest = (
   });
 };
 
+export const getAuthenticationHeaders = async () => {
+  let token = await authenticationPromise;
+
+  if (!token) {
+    throw new Error("Token is mandatory");
+  }
+
+  // Same calc in the jso lib
```

Comment:
If this is truly the same calc that is done in the JSO lib, should we refactor it into a utility function to keep it more DRY.

Reply:
I did change a bit more because the `api/authenticatedRequest` method could benefit as well from the new `api/getAuthenticationHeaders`.

Reply:
We already have a `isJson` check in the request function and you can pass additional headers to authenticatedRequest. Why not just fix the systematic json stringify in the request function and use authenticatedRequest in your class?

Reply:
@ydeshayes Good point. I added a check so In case the body is an instance of FormData, it won't be parsed and the content-type header won't be added by default.

Reply:
thx @MiguelSanchezGonzalez